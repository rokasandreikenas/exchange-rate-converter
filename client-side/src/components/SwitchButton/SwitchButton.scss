@import "../../scss/colors";

.symbol-container {
  display: flex;
  height: 100px;
  width: 100px;
  border-radius: 50%;
  padding: 10px;
  margin-top: 10px;
  align-items: center;
  background: $equal-first-color;
  cursor: pointer;

  .symbol {
    display: flex;
    min-height: 100%;
    min-width: 100%;
    border-radius: 50%;
    background: $equal-second-color;
    align-items: center;
    justify-content: center;

    &:hover {
      background: $equal-first-color;
    }

    i {
      font-size: 24px;
      color: $white-color;
    }
  }
}
