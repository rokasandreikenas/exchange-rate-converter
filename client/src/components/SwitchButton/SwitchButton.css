.symbol-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  height: 100px;
  width: 100px;
  border-radius: 50%;
  padding: 10px;
  margin-top: 10px;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  background: #4987df59;
  cursor: pointer;
}

.symbol-container .symbol {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  min-height: 100%;
  min-width: 100%;
  border-radius: 50%;
  background: #4987df;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.symbol-container .symbol:hover {
  background: #4987df59;
}

.symbol-container .symbol i {
  font-size: 24px;
  color: #ffffff;
}
/*# sourceMappingURL=SwitchButton.css.map */