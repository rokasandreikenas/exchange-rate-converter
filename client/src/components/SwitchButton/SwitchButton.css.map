{
    "version": 3,
    "mappings": "AAEA,AAAA,iBAAiB,CAAC;EAChB,OAAO,EAAE,IAAI;EACb,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,KAAK;EACZ,aAAa,EAAE,GAAG;EAClB,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,IAAI;EAChB,WAAW,EAAE,MAAM;EACnB,UAAU,ECVQ,SAAS;EDW3B,MAAM,EAAE,OAAO;CAoBhB;;AA7BD,AAWE,iBAXe,CAWf,OAAO,CAAC;EACN,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,IAAI;EAChB,SAAS,EAAE,IAAI;EACf,aAAa,EAAE,GAAG;EAClB,UAAU,ECjBO,OAAO;EDkBxB,WAAW,EAAE,MAAM;EACnB,eAAe,EAAE,MAAM;CAUxB;;AA5BH,AAoBI,iBApBa,CAWf,OAAO,AASJ,MAAM,CAAC;EACN,UAAU,ECvBI,SAAS;CDwBxB;;AAtBL,AAwBI,iBAxBa,CAWf,OAAO,CAaL,CAAC,CAAC;EACA,SAAS,EAAE,IAAI;EACf,KAAK,ECzBG,OAAO;CD0BhB",
    "sources": [
        "SwitchButton.scss",
        "../../scss/_colors.scss"
    ],
    "names": [],
    "file": "SwitchButton.css"
}