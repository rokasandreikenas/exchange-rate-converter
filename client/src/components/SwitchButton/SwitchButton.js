import React from "react";
import "./SwitchButton.scss";

const SwitchButton = ({ onClick }) => {
  return (
    <div className="symbol-container" onClick={onClick}>
      <div className="symbol">
        <i className="fas fa-exchange-alt"></i>
      </div>
    </div>
  );
};

export default SwitchButton;
