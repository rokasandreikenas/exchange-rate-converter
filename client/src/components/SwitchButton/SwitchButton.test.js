import React from "react";
import SwitchButton from "./SwitchButton";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";

describe("SwitchButton", () => {
  it("renders SwitchButton component correctly ", () => {
    const output = shallow(<SwitchButton onClick={() => {}} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
