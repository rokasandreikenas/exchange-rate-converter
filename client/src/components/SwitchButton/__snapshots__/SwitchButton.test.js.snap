// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton renders SwitchButton component correctly  1`] = `
<div
  className="symbol-container"
  onClick={[Function]}
>
  <div
    className="symbol"
  >
    <i
      className="fas fa-exchange-alt"
    />
  </div>
</div>
`;
